### DISCUSSION TITLE
/api/v2/search - Not getting expected results

### DISCUSSION THREAD

**Discussion by diaghilev** [mastodon/mastodon#23459] [Q&A] [answered]
My objective is to search Mastodon statuses for a keyword and return the **content** (i.e. the text) of any status containing that keyword. [The docs suggest I can do this](https://docs.joinmastodon.org/methods/search/#200-ok) Can I actually do this?

My python code includes 
```
params = {'q':'myKeyword', 'type':'statuses'}.
```

Yet, I get the following response (even when myKeyword clearly appears in a recent status on my server):
```
{'accounts': [], 'statuses': [], 'hashtags': []}
```
If I don't search specifically for statuses, I get a dictionary like below but no **content** back.
```
{'accounts': [], 'statuses': [], 'hashtags': [{'name': 'hiring', 'url': 'https://data-folks.masto.host/tags/hiring', 'history': [{'day': '1675814400', 'accounts': '5', 'uses': '5'}, {'day': '1675728000', 'accounts': '10', 'uses': '13'}, {'day': '1675641600', 'accounts': '7', 'uses': '7'}, {'day': '1675555200', 'accounts': '6', 'uses': '6'}, {'day': '1675468800', 'accounts': '3', 'uses': '3'}, {'day': '1675382400', 'accounts': '5', 'uses': '6'}, {'day': '1675296000', 'accounts': '9', 'uses': '9'}], 'following': False}]}
```
Thanks for any help! Truly appreciate it.

**Comment by trwnh** [comment, marked as answer]
Full-text search across all statuses is not supported. If your server has configured the optional Elasticsearch backend, then you can perform *limited* full-text search against your own posts, favourites, and bookmarks -- basically, only posts relevant to you. To obtain content based on a keyword, you must use hashtags.

**Reply by diaghilev** [reply]
Thanks! After some searching, I was afraid of that! Appreciate the details.
